src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Dsa.cs

```diff
@@ -65,7 +65,7 @@ internal static int DsaKeySize(SafeDsaHandle dsa)
             // there were some leading zero bytes truncated due to BIGNUM semantics.
             // This minimizes the change the key size is wrong due to leading zero bits; there is only
             // a 2^64 chance that the key has 64 leading zero bits and thus the keysize will be wrong.
```

Comment:
Because p has to be between 2<sup>L-1</sup>and 2<sup>L</sup> the high bit is guaranteed set. And since it's prime, and not the number 2, the low bit is also guaranteed set.

This formula is rounding up to the next byte (in bits), not to a 64-bit boundary:
f(7) => (7 + 7) / 8 * 8 => 14 / 8 * 8 => 1 * 8 => 8.
f(8) => (8 + 7) / 8 * 8 => 15 / 8 * 8 => 1 * 8 => 8.
f(9) => (9 + 7) / 8 * 8 => 16 / 8 * 8 => 2 * 8 => 16.